### ISSUE TITLE
Analytics: Stock Report Table

### ISSUE THREAD

**Issue by timmyc** [woocommerce/wc-admin#920]
**Design Post**: p6riRB-3dV-p2
**Mock API**: NA uses existing endpoints modified for `v3`
**REST API Post**: p7bje6-16i-p2

This issue involves creating the table for the stock report. While this appears to be one of the easier tables we will implement, there are still a few challenges:

- How do we handle retrieving the full parent product name for variations? Additional request to the REST API for certain parent product ID's?
- Table Summary: Currently there isn't a "stats" style endpoint for stock that would return this totals. So that might need to be handled in a follow-up task.

**Comment by Aljullu** [comment]
> Table Summary: Currently there isn't a "stats" style endpoint for stock that would return this totals. So that might need to be handled in a follow-up task.

I'm currently working on that and I'm not sure which solution can be found with the current endpoints. My idea was to make four consecutive calls to the `report` endpoint changing the `type` parameter between `all`, `in_stock`, `low_stock` and `out_of_stock`, and setting `per_page=0` so no results are returned but we just read the `X-WP-Totals` header of each response.

But it looks like `0` is not an allowed value for the `per_page` parameter, and making four API calls just to fill the summary seems a bit overwhelming to me. So that's probably not the way to go.

Any ideas?

**Comment by timmyc** [comment]
Yeah that is entirely far from optimal to make 4 requests to build out the summary. For Store on WPCOM, we created a custom "counts" [endpoint](https://github.com/Automattic/wc-calypso-bridge/blob/master/store-on-wpcom/api/class-wc-calypso-bridge-data-counts-controller.php) to solve this issue - but that endpoint was deemed a bit to app-specific for inclusion as a core Woo endpoint.

One possible approach is to pre-fetch some of these stock numbers and populate in the `wc.` global js namespace. I could see these counts being potentially useful for other UI components. But due to the nature of inventory counts fluctuating often with orders being placed, this could result in some very stale data.

This long comment by yours truly, matched up with my original post makes me think we do indeed need to make this a follow-up task and determine what the best solution is for getting these numbers via the API. Still seems line we need a stats endpoint.

Can we proceed with the rest of this task of creating the table until then?

**Comment by Aljullu** [comment]
> Can we proceed with the rest of this task of creating the table until then?

Sure, we can! Given that we don't have a SwaggerHub endpoint for `stock`, we can't merge the UI before the API is ready, but once #974 gets merged into `master`, I will open #977 for review. And we can let the table summary for a follow-up.
